### ISSUE TITLE
UNDO, THROW - False positive?

### ISSUE THREAD

**Issue by hutcj** [Riverside-Software/sonar-openedge#783]
Regarding this rule:

UNDO statement without option: This rule raises an issue whenever it finds a UNDO keyword without RETRY, LEAVE, UNDO or NEXT (and the default behavior will be to use RETRY).

Shouldn't a THROW be considered a valid option to handle the UNDO?

This rule flagged an UNDO, THROW statement as a violation of this.

![image](https://user-images.githubusercontent.com/40180018/78997089-eda79780-7b13-11ea-84b2-68d9a1783acf.png)

**Comment by gquerret** [comment]
I confirm this is a false positive

**Comment by gquerret** [comment]
What is your code block ? This simple block:

```
do on error undo, throw:

end.
```

Doesn't raise the issue

**Comment by hutcj** [comment]
![image](https://user-images.githubusercontent.com/40180018/79255366-2c12be80-7e54-11ea-813e-aae5e9bc6869.png)

**Comment by stefandrissen** [comment]
You are hiding something in lines 1 - 38, since.

    on error undo, throw:
       message 1.
    end.   

results in:

    ** Unable to understand after -- "on". (247)
    **  Could not understand line 1. (196)

**Comment by hutcj** [comment]
My bad. This is actually a for each loop. Looks like this:

```
for each table1...,
first table2...
on error undo, throw:
...
end.
```

**Comment by gquerret** [comment]
Which version are you using ? I can't reproduce on 2.7, this code doesn't raise any issue:

```
for each customer, first order of customer on error undo, throw:
  message 'hello'.
end.
```

**Comment by hutcj** [comment]
Ok, let me do some testing to see if I can find better replication steps.

**Comment by hutcj** [comment]
Haven't found the steps yet, but this is low priority for us - I'll reopen later once I have time to investigate further.
